demos/enclave/ETL/compile_models.py

```diff
@@ -0,0 +1,335 @@
+            "hispanic_or_latino_ethnicity":
+                str(oneroster_demographics["hispanicOrLatinoEthnicity"][i]).lower()
+        }
+        demographics.append(Demographic.parse_obj(data))
+    return pd.DataFrame([s.__dict__ for s in demographics])
+
+
+def make_name2assessmentID(cli_grades):
+    mapping = {}
+    count = 0
+    for classroom in cli_grades.keys():
+        for user_item in cli_grades[classroom]["usergrades"]:
+            for grade_item in user_item["gradeitems"]:
+                if grade_item["itemname"] is not None:
+                    name = grade_item["itemname"].strip('"')
+                    if name not in mapping.keys():
+                        mapping[name] = count
+                        count += 1
+    return mapping
+
+def make_email2uuid(cli_users):
+    mapping = {}
+    for classroom in cli_users.keys():
+        for user in cli_users[classroom]:
+            mapping[user["email"]] = uuid.uuid4()
+    return mapping 
+
+def make_userid2email(cli_users):
+    mapping = {}
+    for classroom in cli_users.keys():
+        for user in cli_users[classroom]:
+            mapping[user["id"]] = user["email"]
+    return mapping
+
+def make_sourceid2email(oneroster_users):
+    mapping = {}
+    for i in range(0, len(oneroster_users['sourceId'])):
+        if oneroster_users["sourceId"][i] != "nan":            
+            mapping[oneroster_users["sourceId"][i]] = oneroster_users["email"][i]
+    return mapping
+
+
+def create_models(
+    output_path, cli_grades, cli_users, oneroster_demographics, oneroster_users, 
+):
+    sourceid_2_email = make_sourceid2email(oneroster_users)
+    userid_2_email = make_userid2email(cli_users)
+    email_2_uuid = make_email2uuid(cli_users)
+    name_2_assessmentId = make_name2assessmentID(cli_grades)
+
+    demographics_df = demographics_model(oneroster_demographics,sourceid_2_email, email_2_uuid)
+    users_df = users_model(oneroster_users, sourceid_2_email, email_2_uuid)
+    grades_df = grades_model(cli_grades, userid_2_email, email_2_uuid, name_2_assessmentId)
+    enrollments_df = enrollments_model(cli_users, email_2_uuid)
+    courses_df = courses_model(cli_users)
+    assessments_df = assessment_model(name_2_assessmentId)
+
+    with open(f"{output_path}/demographics.csv", "w") as f:
+        demographics_df.to_csv(f, index=False, header=True)
+    with open(f"{output_path}/users.csv", "w") as f:
+        users_df.to_csv(f, index=False, header=True)
+    with open(f"{output_path}/grades.csv", "w") as f:
+        grades_df.to_csv(f, index=False, header=True)
+    with open(f"{output_path}/enrollments.csv", "w") as f:
+        enrollments_df.to_csv(f, index=False, header=True)
+    with open(f"{output_path}/courses.csv", "w") as f:
+        courses_df.to_csv(f, index=False, header=True)
+    with open(f"{output_path}/assessments.csv", "w") as f:
+        assessments_df.to_csv(f, index=False, header=True, quoting=QUOTE_NONE, escapechar='\\')
```

Comment:
I remembered us talking about commas being escaped with a '\' earlier (which is what this line should do) - otherwise the whole title will be put in quotes in the csv if it has a comma. Happy to remove this.

Reply:
Oh, that may have been a misunderstanding. I  meant to convey that these libraries will escape things as necessary. If you remove this you'll see the entire value is put into quotes which works / prevents a CSV reader from parsing on the commas in the string. In general, I don't think we should need to do anything special if we're using either the `pandas` or `csv.DictWriter` serializers.